### PR TITLE
feat(spell-check): added spell check to the pipeline

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction ✏️ 
Ensure spelling check is performed as part of the pipeline.

## Resolution ✔️ 
* Added `spellcheck` service to be executed during CICD under SCA.
* Spelling fixes.
* File rename.

## Miscellaneous ➕ 
* Dependencies updates.
* Removed `npm-check-updates` dependency.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,14 +21,15 @@
     "lint:all": "npx eslint ./azure-functions/acbs-function && npx eslint ./azure-functions/number-generator-function && npx eslint ./dtfs-central-api && npx eslint ./e2e-tests && npx eslint ./gef-ui && npx eslint ./portal && npx eslint ./portal-api && npx eslint ./external-api && npx eslint ./trade-finance-manager-api && npx eslint ./trade-finance-manager-ui && npx eslint ./utils/mock-data-loader && npx eslint ./utils/data-migration",
     "lint:fix:all": "npx eslint --fix ./azure-functions/acbs-function && npx eslint --fix ./azure-functions/number-generator-function && npx eslint --fix ./dtfs-central-api && npx eslint --fix ./e2e-tests && npx eslint --fix ./gef-ui && npx eslint --fix ./portal && npx eslint --fix ./portal-api && npx eslint --fix ./external-api && npx eslint --fix ./trade-finance-manager-api && npx eslint --fix ./trade-finance-manager-ui && npx eslint --fix ./utils/mock-data-loader && npx eslint --fix ./utils/data-migration",
     "prepare": "husky install",
+    "spellcheck": "cspell lint --gitignore --no-must-find-files --unique --no-progress --show-suggestions --color '**/*'",
```
Comment (lmarrai-sw) [inline package.json:24]: I think it's going to find a lot of spelling issues in old files (until now, we've only been spell-checking files that we've edited) :)

**Review by lmarrai-sw — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (abhi-markan) [reply package.json:24]: Yep around ~1,700 issues I need to resolve.

**Review by lmarrai-sw — COMMENTED** [review]
That's some real dedication to solve all of these typos!

File: external-api/api-tests/v1/ordnance-survey.api-test.ts [hunk 2]
```diff
@@ -41,7 +41,7 @@ const mockResponse = {
           POSTAL_ADDRESS_CODE_DESCRIPTION: 'A record which is linked to PAF',
           BLPU_STATE_CODE: null,
           BLPU_STATE_CODE_DESCRIPTION: 'Unknown/Not applicable',
-          TOPOGRAPHY_LAYER_TOID: 'osgb1000019181070',
+          TOPOGRAPHY_LAYER_TOID: 'mock',
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply external-api/api-tests/v1/ordnance-survey.api-test.ts:44]: Good spot, have rectified.

File: external-api/api-tests/v1/ordnance-survey.api-test.ts [hunk 3]
```diff
@@ -71,7 +71,7 @@ const mockResponse = {
           POSTAL_ADDRESS_CODE_DESCRIPTION: 'A record which is linked to PAF',
           BLPU_STATE_CODE: null,
           BLPU_STATE_CODE_DESCRIPTION: 'Unknown/Not applicable',
-          TOPOGRAPHY_LAYER_TOID: 'osgb1000019180992',
+          TOPOGRAPHY_LAYER_TOID: 'mock',
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply external-api/api-tests/v1/ordnance-survey.api-test.ts:74]: Good spot, have rectified.

File: package.json [hunk 4]
```diff
@@ -16,11 +16,11 @@
...
     "ci:all": "npm ci --legacy-peer-deps && npm ci --legacy-peer-deps --prefix azure-functions/acbs-function && npm ci --legacy-peer-deps --prefix azure-functions/number-generator-function && npm ci --legacy-peer-deps --prefix dtfs-central-api && npm ci --prefix gef-ui && npm ci --legacy-peer-deps --prefix portal && npm ci --legacy-peer-deps --prefix portal-api && npm ci --legacy-peer-deps --prefix external-api && npm ci --legacy-peer-deps --prefix trade-finance-manager-api && npm ci --legacy-peer-deps --prefix trade-finance-manager-ui && npm ci --legacy-peer-deps --prefix utils/mock-data-loader && npm ci --legacy-peer-deps --prefix utils/data-migration",
-    "housekeeping": "npm run update:all && npm run install:all && npm run lint:fix:all && npm run audit:fix:all",
+    "housekeeping": "npm run update:all && npm run install:all && npm run lint:fix:all && npm run audit:fix:all && npm run spellcheck",
+    "postinstall": "husky install",
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply package.json:20]: Wanted to ensure `husky` is ready after dependency installation.

File: portal-api/api-tests/drivers/fileshare.api-test.js [hunk 5]
```diff
@@ -8,8 +8,8 @@ const someXML2 = '<?xml version="1.0" encoding="UTF-8"?>XML2<Deal/>';
 const folder = 'api_tests/fileshare';
 const fileshareName = 'portal';
 
-// Fileshare tests are not picking up Azure env variables. Disbabling for time being
-xdescribe('fileshare', () => {
+// Fileshare tests are not picking up Azure env variables. Disabling for time being
+describe('fileshare', () => {
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply portal-api/api-tests/drivers/fileshare.api-test.js:12]: Good idea, have used.

File: portal/server/controllers/dashboard/filters/generate-keyword-query.js [hunk 6]
```diff
@@ -4,11 +4,11 @@
  * @param {string} fieldName
  * @param {string} keyword value
  * @example ( 'submissionType', 'Automatic' )
- * @returns { dealType: { REGEX: 'Automatic', $options: 'i' } }
+ * @returns { dealType: { REGEX: 'Automatic', $options: 'mock' } }
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply portal/server/controllers/dashboard/filters/generate-keyword-query.js:7]: Reverted.

File: portal/server/controllers/dashboard/filters/generate-keyword-query.js [hunk 7]
```diff
@@ -4,11 +4,11 @@
...
  */
 const generateObject = (fieldName, keywordValue) => ({
   [fieldName]: {
-    REGEX: keywordValue, $options: 'i',
+    REGEX: keywordValue, $options: 'mock',
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply portal/server/controllers/dashboard/filters/generate-keyword-query.js:11]: Reverted.

File: portal/server/controllers/dashboard/filters/generate-keyword-query.js [hunk 8]
```diff
@@ -18,7 +18,7 @@ const generateObject = (fieldName, keywordValue) => ({
  * @param {array} fields
  * @param {string} keyword value
  * @example ( ['dealType', 'submissionType' ], 'Automatic' )
- * @returns [ { dealType: { REGEX: 'Automatic', $options: 'i' } }, { submissionType: { REGEX: 'Automatic', $options: 'i' } } ]
+ * @returns [ { dealType: { REGEX: 'Automatic', $options: 'mock' } }, { submissionType: { REGEX: 'Automatic', $options: 'mock' } } ]
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply portal/server/controllers/dashboard/filters/generate-keyword-query.js:21]: Reverted.

File: portal/server/controllers/dashboard/filters/generate-keyword-query.test.js [hunk 9]
```diff
@@ -17,7 +17,7 @@ describe('controllers/dashboard/filters - generate-keyword-query', () => {
 
       const expected = {
         [mockFieldName]: {
-          REGEX: mockKeywordValue, $options: 'i',
+          REGEX: mockKeywordValue, $options: 'mock',
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply portal/server/controllers/dashboard/filters/generate-keyword-query.test.js:20]: Reverted.

File: trade-finance-manager-api/src/constants/facilities.js [hunk 10]
```diff
@@ -32,7 +32,7 @@ const FACILITY_FEE_TYPE_PORTAL = {
 
 const FACILITY_FEE_TYPE_GEF = {
   IN_ARREARS_MONTHLY: 'IN_ARREARS_MONTHLY',
-  IN_ARREARS_QUARTLY: 'IN_ARREARS_QUARTLY',
+  IN_ARREARS_QUARTERLY: 'IN_ARREARS_QUARTERLY',
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply trade-finance-manager-api/src/constants/facilities.js:35]: The only value stored in DB is `In arrear`, above should not effect any processing.

File: trade-finance-manager-api/src/v1/__mocks__/mock-deal-no-party-db.js [hunk 11]
```diff
@@ -167,7 +167,7 @@ const MOCK_DEAL_NO_PARTY_DB = {
   bondTransactions: {
     items: [
       {
-        _id: '61f7a4edcf809301e78fbg43',
+        _id: 'mock',
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply trade-finance-manager-api/src/v1/__mocks__/mock-deal-no-party-db.js:170]: Good spot, have reverted back to valid test Mongo ID.

File: utils/data-migration/constant/facility.js [hunk 12]
```diff
@@ -32,7 +32,7 @@ const FACILITY_FEE_TYPE_PORTAL = {
 
 const FACILITY_FEE_TYPE_GEF = {
   IN_ARREARS_MONTHLY: 'IN_ARREARS_MONTHLY',
-  IN_ARREARS_QUARTLY: 'IN_ARREARS_QUARTLY',
+  IN_ARREARS_QUARTERLY: 'IN_ARREARS_QUARTERLY',
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply utils/data-migration/constant/facility.js:35]: Same as above.

File: utils/data-migration/gef/map-deal.js [hunk 13]
```diff
@@ -225,13 +225,13 @@ const mapV1Deal = async (token, v1Deal, v2Banks, v2Users) => {
     createdAt: convertDateToTimestamp(v1Deal.created),
     updatedAt: convertDateToTimestamp(v1Deal.changed),
     bank: getBankByName(v2Banks, v1Deal.owner.bank),
-    mandatoryVersionId: Number(v1Deal.children.eligiblity.system_red_line_revision_id),
+    mandatoryVersionId: Number(v1Deal.children.eligibility.system_red_line_revision_id),
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply utils/data-migration/gef/map-deal.js:228]: This script is no longer in use and it should not break any deals.

**Review by abhi-markan — COMMENTED** [review]
Have pushed changes @lmarrai-sw

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (lmarrai-sw) [reply package.json:20]: Ah okay, I think that already worked with `prepare` but `postinstall` is fine too

**Review by lmarrai-sw — APPROVED** [review]

**Pull request merged**